src/main/java/thePackmaster/patches/warlockpack/ImpGameActionManagerPatch.java

```diff
@@ -0,0 +1,33 @@
+package thePackmaster.patches.warlockpack;
+
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.GameActionManager;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import thePackmaster.cards.warlockpack.Imp;
+
+// HandCardSelectScreen calls GameActionManager.cleanCardQueue, and is one of only two places that calls that method, the
+// other being AbstractMonster.die(). If you draw imps at the same time as a card that opens HandCardSelectScreen is played,
+// this means that the card queue item for playing the imps vanishes. To fix this, we add them back in an action that
+// executes afterwards.
+@SpirePatch(clz = GameActionManager.class, method = "cleanCardQueue", paramtypez = {})
+public class ImpGameActionManagerPatch {
+    @SpirePostfixPatch
+    public static void addImpsBackToQueue() {
+        if (AbstractDungeon.currMapNode != null && AbstractDungeon.currMapNode.room != null && AbstractDungeon.currMapNode.room.monsters != null && !AbstractDungeon.currMapNode.room.monsters.areMonstersBasicallyDead()) {
```

Comment:
This series of checks may be more paranoid than necessary, but the core reason for having it is for the `areMonstersBasicallyDead` check at the end. That's because `cleanCardQueue` is called from `AbstractMonster.die` when `areMonstersBasicallyDead` is `true`, and in that case, we don't want to add imps back to the queue (since combat is over).